src/plugins/ucx/ucx_backend.cpp

```diff
@@ -360,14 +361,12 @@ void nixlUcxEngine::progressFunc()
     }
     pthrActiveCV.notify_one();
 
-    // Set POLLIN event on all worker fds so that the main loop would process them all on first iteration
-    for (size_t wid = 0; wid < pollFds.size() - 1; wid++)
-        pollFds[wid].revents = POLLIN;
-
+    // Set timeout event so that the main loop would progress all workers on first iteration
+    bool timeout = true;
     bool pthrStop = false;
     while (!pthrStop) {
         for (size_t wid = 0; wid < pollFds.size() - 1; wid++) {
-            if (!(pollFds[wid].revents & POLLIN))
+            if (!(pollFds[wid].revents & POLLIN) && !timeout)
```

Comment:
it seems like logic around `!timeout` condition can be simplified, like moving the condition outside of `for(...)` loop

Reply:
Not sure I got you. How would moving this conditional outside of for loop look like?

Reply:
sorry, I missed that the `!timeout` is a 2nd condition in the statement